GUI/Configuration.cs

```diff
@@ -32,7 +33,25 @@ public class Configuration
         public int SortByColumnIndex = 2;
         public bool SortDescending = false;
 
-        public bool[] VisibleColumns = { true, true, true, true, true, true, true, true, true };
+        [XmlArray, XmlArrayItem(ElementName = "ColumnName")]
+        public List<string> VisibleColumnNames = new List<string>();
```

Comment:
If we add a new column in the future, it will be hidden first, until the user manually unhides it, right?
What if we don't store the visible, but the hidden columns in the array? This way potential new columns should be shown after an update.

Reply:
Hmm, good point. I'll think about this and get back to you.

Reply:
OK, storing HiddenColumnNames instead works pretty well and also significantly simplifies the defaulting mechanism, since we no longer need to maintain that list of column names (empty = all visible). However, it also means that we probably would not be able to use such a setting for column reordering in the future; if all we know is that a column isn't hidden, that doesn't tell us which index it should occupy.

Can you think of any other way we could dynamically add new columns to the visible list while still allowing the user to remove them? Maybe a `List<KeyValuePair<string, bool>>`, with missing columns treated as visible?

Reply:
Maybe the best option is to implement HiddenColumnNames now, and then in the future we can add a VisibleColumnNames list for ordering. That way if a column name isn't in either one, we know it should be added to the visible list. I'll go with that for now.

Reply:
Sounds good :+1: